master/getting-started/kubernetes/installation/hardway/install-cni-plugin.md

```diff
@@ -0,0 +1,184 @@
+           -nodes \
+           -out cni.csr \
+           -subj "/CN=calico-cni"
+```
+
+We will sign this certificate using the main Kubernetes CA.
+
+```
+sudo openssl x509 -req -in cni.csr \
+                  -CA /etc/kubernetes/pki/ca.crt \
+                  -CAkey /etc/kubernetes/pki/ca.key \
+                  -CAcreateserial \
+                  -out cni.crt \
+                  -days 365
+sudo chown ubuntu:ubuntu cni.crt
+```
+
+Next, we create a kubeconfig file for the CNI.
+
+```
+APISERVER=$(kubectl config view -o jsonpath='{.clusters[0].cluster.server}')
+kubectl config set-cluster kubernetes \
+    --certificate-authority=/etc/kubernetes/pki/ca.crt \
+    --embed-certs=true \
+    --server=$APISERVER \
+    --kubeconfig=cni.kubeconfig
+
+kubectl config set-credentials calico-cni \
+    --client-certificate=cni.crt \
+    --client-key=cni.key \
+    --embed-certs=true \
+    --kubeconfig=cni.kubeconfig
+
+kubectl config set-context default \
+    --cluster=kubernetes \
+    --user=calico-cni \
+    --kubeconfig=cni.kubeconfig
+
+kubectl config use-context default --kubeconfig=cni.kubeconfig
+```
+
+Copy this `cni.kubeconfig` file to every node in the cluster.
+
+## Provision RBAC
+
+Define a ClusterRole the CNI Plugin will use to access Kubernetes.
+
+```
+kubectl apply -f - <<EOF 
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: calico-cni
+rules:
+  # The CNI plugin needs to get pods, nodes, and namespaces.
+  - apiGroups: [""]
+    resources:
+      - pods
+      - nodes
+      - namespaces
+    verbs:
+      - get
+  # The CNI plugin patches pods/status.
+  - apiGroups: [""]
+    resources:
+      - pods/status
+    verbs:
+      - patch
+ # These permissions are required for Calico CNI to perform IPAM allocations.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - blockaffinities
+      - ipamblocks
+      - ipamhandles
+    verbs:
+      - get
+      - list
+      - create
+      - update
+      - delete
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - ipamconfigs
+      - clusterinformations
+      - ippools
+    verbs:
+      - get
+      - list
+EOF
+```
+
+Bind the ClusterRole to the `calico-cni` account.
+
+```
+kubectl create clusterrolebinding calico-cni --clusterrole=calico-cni --user=calico-cni
+```
+
+## Install the plug in
+
+Do these steps on each node in your cluster.
+
+Run these commands as root.
+
+```
+sudo su
+```
+
+Install the CNI Plugin Binaries
+
+```
+curl -L -o /opt/cni/bin/calico https://github.com/projectcalico/cni-plugin/releases/download/v3.8.0/calico-amd64
+chmod 755 /opt/cni/bin/calico
+curl -L -o /opt/cni/bin/calico-ipam https://github.com/projectcalico/cni-plugin/releases/download/v3.8.0/calico-ipam-amd64
+chmod 755 /opt/cni/bin/calico-ipam
+```
+
+Create the config directory
+
+```
+mkdir -p /etc/cni/net.d/
+```
+
+Copy the kubeconfig from the previous section
+``` 
+cp cni.kubeconfig /etc/cni/net.d/calico-kubeconfig
```

Comment:
Any reason we rename the file? Might be less cognitive overload for a reader if it was called `calico-kubeconfig` from the start.

Reply:
It's `cni.kubeconfig` to distinguish it from other kubeconfigs we write in other parts of the guide. calico-kubeconfig is pretty generic as a name.